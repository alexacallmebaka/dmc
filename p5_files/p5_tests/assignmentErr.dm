a:int;
b:bool;

f : () void {
	a = b;
}
C : class {
};
main : () int {
  a : perfect int = 5;
  take f;
  take a;
	b : int;
	b = f;
	b = C;
}