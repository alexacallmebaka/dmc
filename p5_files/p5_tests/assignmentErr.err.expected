FATAL [5,2]-[5,7]: Invalid assignment operation
FATAL [11,8]-[11,9]: Attempt to assign user input to function
FATAL [12,10]-[12,11]: Non-lval assignment
FATAL [14,6]-[14,7]: Invalid assignment operand
FATAL [15,6]-[15,7]: Invalid assignment operand
Type Analysis Failed
